integration/full_stack_test.go

```diff
@@ -166,6 +164,19 @@ func verifyBuildFilesExist(t *testing.T, buildPath string) {
 	}
 }
 
+func waitUntilAvailable(addr string) {
+	for {
+		conn, err := net.Dial("tcp", addr)
+		if conn != nil {
+			// Silently ignore the err
+			_ = conn.Close()
```

Comment:
maybe this here, so we don't bombard the test server with connection attempts?
```suggestion
			_ = conn.Close()
			time.Sleep(time.Second)
```

Reply:
Also, maybe we could add a timer that fails after x tries (like 10 tries?)

Reply:
D'oh, of course.

Reply:
> Also, maybe we could add a timer that fails after x tries (like 10 tries?)
I don't want to worry about that.  Tests have a top-level time limit.